packages/nextjs/src/adapter-nextjs/page-props/story.ts

```diff
@@ -22,19 +22,30 @@ export function getStoryPageServerSideProps<CustomProps extends Record<string, a
     return async function getServerSideProps(
         context: GetServerSidePropsContext,
     ): Promise<GetServerSidePropsResult<CustomProps>> {
-        const api = NextContentDelivery.initClient(context.req, { formats });
+        const { locale, req, params } = context;
 
-        const { slug } = context.params as { slug?: string };
+        const api = NextContentDelivery.initClient(req, { formats });
+
+        const { slug } = params as { slug?: string };
         const story = slug ? await api.story({ slug }) : null;
         if (!story) {
             return { notFound: true };
         }
 
+        // [DEV-12082] If the current URL pathname does not match the canonical URL -- redirect.
+        if (req.url && new URL(req.url).pathname !== `/${story.slug}`) {
+            return {
+                redirect: {
+                    destination: `/${slug}`,
```

Comment:
I believe it should redirect to `/${story.slug}` 🤔

Reply:
Thank you Oleg! :+1: